public/app/features/variables/shared/multiOptions.ts

```diff
@@ -0,0 +1,39 @@
+import { VariableOption } from 'app/features/templating/types';
+
+export const alignCurrentWithMulti = (current: VariableOption, value: boolean): VariableOption => {
+  if (!current) {
+    return current;
+  }
+
+  if (value && !Array.isArray(current.value)) {
+    return {
+      ...current,
+      value: convertToMulti(current.value),
+      text: convertToMulti(current.text),
+    };
+  }
+
+  if (!value && Array.isArray(current.value)) {
+    return {
+      ...current,
+      value: convertToSingle(current.value),
+      text: convertToSingle(current.text),
+    };
+  }
+
+  return current;
+};
+
+const convertToSingle = (value: string | string[]): string => {
+  if (Array.isArray(value) && value.length > 0) {
+    return value[0];
+  }
+  return value.toString();
```

Comment:
Hmm, do we need the toString()?

Reply:
hmm, we might change it to `return ''` because it can happen if it is an array that is empty. Or if it isn't an array.